processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -55,6 +61,18 @@ public static void verifyAggregations(
             "Missing fields [%s] for postAggregator [%s]", missing, postAgg.getName()
         );
         Preconditions.checkArgument(combinedAggNames.add(postAgg.getName()), "[%s] already defined", postAgg.getName());
+
+        if (postAgg instanceof HasDependentAggFactories) {
```

Comment:
@nishantmonu51 You're right. Injections of post aggs in Arithmetic post aggregator does not work because jackson injects the post aggs while creating an instance which means there is no chance to go through Queries.prepareAggregations()